crates/turborepo-lib/src/commands/link.rs

```diff
@@ -127,56 +126,125 @@ pub async fn link(base: &mut CommandBase, modify_gitignore: bool) -> Result<()>
         )
     })?;
 
-    let teams_response = api_client
-        .get_teams(token)
-        .await
-        .context("could not get team information")?;
+    match target {
+        LinkTarget::RemoteCache => {
```

Comment:
These `match` arms are pretty big. May be a good idea to split them into separate functions?